src/core/memory.cc

```diff
@@ -1890,7 +1890,9 @@ void* realloc(void* ptr, size_t size) {
     if (try_trigger_error_injector()) {
         return nullptr;
     }
-    if (ptr != nullptr && !is_seastar_memory(ptr)) {
+    if (ptr == nullptr) {
+        return malloc(size);
+    } else if (!is_seastar_memory(ptr)) {
```

Comment:
below there's the following comment:

"// if we're here, it's either ptr is a seastar memory ptr
    // or a nullptr, or, original functions aren't available"

we've to update it to "if we're here, it's a non-null seastar memory ptr..."

Reply:
Fixed in v2